XdmodAID.md

```diff
@@ -0,0 +1,868 @@
+With storage metrics, each storage class will have it's own resource.
+
+# xdmod-openshift
+
+This was mainn's constribution to this project.  Currently it runs within the container that
+is built from the Docker.moc-xdmod docker file.  It should be refactored to run in the
+python3.11 container like xdmod-openstack does.
+
+What this script does is to make openshift look like a slurm cluster to xdmod.  The log
+files can be shreadded and ingested in the same manner
+
+My current recommendations (also in the issues),
+
+    1.  move this to a python 3.11 container as it is python only
+    2.  fix the manifests, instead of having a ..._prod and a ..._staging manifest adopt the
+        following structure:
+            i. create a generic manifest within the main directory for xdmod-openshift
+           ii. create specific overlays in each of the supported overlay directires
+
+It wasn't until we actually started looking at the data in May that we realized that
+CPU was not appearning for jobs in the interface.  As this was something that I was very
+focused on in getting data in on the OpenStack side and how I found the flavors that
+were deleted, I was surprised that it wasn't noticed by anyone until we were gathering
+data.  So I started looking in to this.
+
+With kim, we looked at the database structure and the data that was being pulled from
+OpenShift.  The type coming from openshift is in fractions of a CPU which is a floating
+point type, where as the database field that this is stored in is an iteger.  Addtionoally,
+in the supremm database, this becomes a floating point.
+
+I have just started to work with the data that is processed here as it was not inserting any
+CPU data into the jobs table.  This was due to the data coming from openshift often in
+fractional amounts of CPU.  One simple solution to this is to store the data in milli CPU by
+multiplying by 1000.  This was tested in the xdmod-staging project on the nerc-infra cluster
+and seemed to work (ie, I got milli CPU in the jobs table).
+
+I then tried to setup Supremm on the staging project, and the process failed when running
+from xdmod-setup. I did get the staging mariadb setup by restoring from a backup of production
+though something is wrong with the configuration of mongo
+
+
+
+# xdmod-hierarchy
+
+In order to report to insititutions, departments and PIs; a system wide hierarchy can
+be defined to aggreate statistics at the appropriate level.  This is done using
+csv files that define each level - the xdmod documentation  explaination can be
+found in (https://open.xdmod.org/10.0/hierarchy.html,
+https://open.xdmod.org/10.0/user-names.html, https://open.xdmod.org/10.0/cloud.html).
+I will try to cosolidate and explain this a bit better.
+
+Depending on if it is a HPC job or a cloud resource there are different levels to
+the hierarchy.  This is due to jobs not having a project name associated with
+them.  To side by side compare:
```

Comment:
decide if you want one or two spaces after periods and then make sure they are all the same.

Reply:
ok, 2 spaces after a single '.'